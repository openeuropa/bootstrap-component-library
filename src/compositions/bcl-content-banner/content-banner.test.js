import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-content-banner/data/data";

const template = "@oe-bcl/bcl-content-banner/content-banner.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Content banner", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});
