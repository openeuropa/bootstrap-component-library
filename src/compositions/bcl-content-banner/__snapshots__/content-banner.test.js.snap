// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Content banner renders correctly 1`] = `
<jest>
  <div
    class="bcl-content-banner vw-100 pt-lg-4-75 pb-md-4-75 pb-4 pt-4 bg-lighter"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="bg-transparent border-0 card"
        >
          <div
            class="row"
          >
            <div
              class="col-md-4 col-lg-3"
            >
              <img
                alt="alt img"
                class="card-img-top"
                src="https://picsum.photos/255/255?random=6"
              />
            </div>
            <div
              class="col-md-8 col-lg-9 col-xxl-8"
            >
              <div
                class="card-body pt-md-0 px-0 px-md-3 pb-0"
              >
                <h2
                  class="fw-bold card-title"
                >
                  This is the title of this page
                </h2>
                <div
                  class="mb-4 mt-4"
                >
                  <span
                    class="text-muted me-3"
                  >
                    8 hours ago
                  </span>
                </div>
                <p>
                  
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse volutpat ultricies massa, a dapibus libero porta nec.
        Sed facilisis dictum vestibulum. Fusce commodo hendrerit diam, pretium tempus leo varius sit amet.
        Etiam interdum, orci at sagittis luctus, lorem libero tempus mauris, a fermentum libero orci semper lacus.
        Duis tristique fringilla magna, eu egestas dolor molestie non.
     
                </p>
                <div
                  class="mt-2-5"
                >
                  <span
                    class="me-2 badge rounded-pill badge-outline-primary"
                  >
                    Category 1
                  </span>
                  <span
                    class="badge rounded-pill badge-outline-primary"
                  >
                    Category 2
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="d-flex justify-content-end"
        >
          <button
            class="w-auto mt-3 me-3 btn btn-light btn-md"
            type="button"
          >
            Print
            <svg
              class="ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="/icons.svg#printer-fill"
              />
            </svg>
          </button>
          <button
            class="w-auto mt-3 btn btn-light btn-md"
            type="button"
          >
            Export
            <svg
              class="ms-2-5 bi icon--fluid"
            >
              <use
                xlink:href="/icons.svg#file-arrow-down-fill"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</jest>
`;
